### PR TITLE
Oh we're ✨~SASSy✨ (Installing Sass)

### LINKED ISSUE #28: 13.2 As a developer, I want to bring my design to life
## Acceptance criteria
- [x] Create the style branch
- [x] ~Install MUI~ Install Sass
- [x] ~Create custom MUI theme~ Add some color & font variables
- [x] Install fonts

### PR DESCRIPTION
**Pull request by itsoliviasparks**
## Description

- Installed sass:
    - `npm install sass`
    - deleted all old css files, made `src/App.scss`, &  added `src/sass` folder to house sass partials
    - made `_variables.scss` to house font-families & colors

- Light / Dark Mode:
    - maintained `<meta name="color-scheme" content="dark light" />` tag in `index.html` to auto choose color mode based on users system preferences
    - used `@media (prefers-color-scheme: dark)` to change colours for light/dark mode

- Styling:
    - installed google fonts
    - made variables for fonts & colours
    - Added wrapper to `<main>`
    - Began stying on header
    - moved nav bar into header
    - Added a footer with links to contributor's GitHub accounts, the repo, & the Collab Lab site

- Misc
    - Deleted some old comments
    - Added a lemon emoji for the favicon
    - Updated the `<title>` in `index.html`

## Related Issue

closes #28 

## Acceptance Criteria
- [x] Create the style branch
- [x] Install Sass
- [x] Add color & font variables
- [x] Install fonts

## Type of Changes
|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|   ✓  | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
Screenshot 2023-05-20 at 14 41 08

### After
**Light Mode:**
Screenshot 2023-05-20 at 14 18 11

**Dark Mode:**
Screenshot 2023-05-20 at 14 15 04

## Testing Steps / QA Criteria

- 👀 just open your browser and have a peak!
- You can turn off dark mode by commenting out the dark mode partial in `src/App.scss` or toggle between modes with your user settings

### CODE REVIEW

File: src/views/AddItem.jsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { React, useState } from 'react';
 
 import { addItem } from '../api/firebase';
 
-// Removes excess spaces, punctuation, special characters, and makes all leters lowercase.
+// Removes excess spaces, punctuation, special characters, and makes all letters lowercase.
```
Comment (lizzypine) [inline src/views/AddItem.jsx:5]: I see you. Nice attention to detail.

File: src/views/Layout.jsx [hunk 2]
```diff
@@ -1,28 +1,39 @@
...
-						Add Item
-					</NavLink>
-				</nav>
+				<div className="wrapper">
+					<main className="Layout-main">
+						<Outlet />
+					</main>
+					<footer>
+						Created by <a href="https://github.com/hansontram">Hanson Tram</a>,{' '}
+						<a href="https://github.com/lizzypine">Lizzie Pine</a>,{' '}
```
Comment (lizzypine) [inline src/views/Layout.jsx]: I'm in the non-traditional Lizzy-with-a-"y" camp. Mind updating the spelling?

File: src/views/Layout.jsx [hunk 3]
```diff
@@ -1,28 +1,39 @@
...
+						<a href="https://github.com/itsoliviasparks">Olivia Sparks</a>,{' '}
+						<a href="https://github.com/01001101CK">Yufa Li</a> & for{' '}
+						<a href="https://the-collab-lab.codes">The Collab Lab</a>.{' '}
+						<a href="https://github.com/the-collab-lab/tcl-61-smart-shopping-list">
+							Browse the REPO
```
Comment (lizzypine) [inline src/views/Layout.jsx:32]: Another nice detail. Thanks for adding in the credits!

**Review by lizzypine — APPROVED** [review]
This looks great! I appreciate the attention to the little things such as the unique favicon and dark/light mode capability. This was the perfect point in time to remove extraneous comments, so thanks for tackling that as you went along. As an aside, with 22 different files changes, I think this PR wins the award for most file changes!

File: src/views/Layout.jsx (hunk 2 above)
Comment (itsoliviasparks) [reply src/views/Layout.jsx]: omg of course! I'm sorry!

File: src/sass/_variables.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+$red-dark: #ff404a;
+$red-light: #ff3644;
+$yellow: #e6f51d;
+$brown: #553743;
+
+//colors dark mode:
+$dark-purple: #241c53;
+$teal: #03dac6;
+$dark-yellow: #eee8a9;
+$pop-purple: #ad7cfc;
```
Comment (01001101CK) [inline src/sass/_variables.scss:19]: Thank you for doing this, so we don't need to refer to the Figma design constantly. :)

File: src/App.jsx [hunk 5]
```diff
@@ -13,41 +13,15 @@ import { useStateWithStorage } from './utils';
...
-	/**
-	 * Here, we're using a custom hook to create `listToken` and a function
-	 * that can be used to update `listToken` later.
-	 *
-	 * `listToken` is `my test list` by default so you can see the list
-	 * of items that was prepopulated for this project.
-	 * You'll later set it to `null` by default (since new users do not
-	 * have tokens), and use `setListToken` when you allow a user
-	 * to create and join a new list.
-	 */
```
Comment (01001101CK) [inline src/App.jsx:25]: Thank you for thinking ahead!  Removing comments as we go is such a good practice.

File: src/sass/_header.scss [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	&:focus {
+		background-color: $yellow;
+		color: $brown;
+	}
+}
+
+.Nav-link.special {
+	background-color: $red-light;
+	border-radius: 50%;
+	font-size: 4.5rem;
```
Comment (01001101CK) [inline src/sass/_header.scss:41]: Love the different usage of `vh` and `rem`. I struggled to learn the difference for a long while!

**Review by 01001101CK — APPROVED** [review]
This looks awesome! We are ready to style our app. Collectively, we made an engineering decision to use SASS instead of Material UI because it's much more customizable and flexible. You are already thinking ahead to remove comments, add credits and add a lemon emoji for the favicon! Nice work.

File: src/sass/_variables.scss (hunk 4 above)
Comment (itsoliviasparks) [reply src/sass/_variables.scss:19]: love a variable!

**Pull request merged**